data/transactions/logic/eval.go

```diff
@@ -1010,12 +1009,16 @@ func (cx *EvalContext) evalError(err error) error {
 	pc, det := cx.pcDetails()
 	details := fmt.Sprintf("pc=%d, opcodes=%s", pc, det)
 
-	return EvalError{
-		serr.Annotate(err, "pc", pc, "group-index", cx.groupIndex,
-			"eval-states", cx.evalStates()),
-		details,
-		cx.runMode == ModeSig,
+	err = basics.Annotate(err,
+		"pc", pc,
+		"group-index", cx.groupIndex,
+		"eval-states", cx.evalStates())
+	if cx.runMode == ModeApp {
+		details = fmt.Sprintf("app=%d %s", cx.appID, details)
+		err = basics.Annotate(err, "app-index", cx.appID)
```

Comment:
nit: the original details string separates pc and opcodes by a comma. For consistency it would be nice to do the same after app, e.g. `app=%d, %s`

Also, why use a different name, `app-index` in the annotation?

Reply:
Definitely agree on the comma.

I used "app" to save some space in what is supposed to be a human readable error. I used "app-index" because that's the form we use in all of our JSON responses.

I'm willing to put app-index in the message if you think I should.